screenpipe-server/src/core.rs

```diff
@@ -32,28 +33,27 @@ pub async fn start_continuous_recording(
     friend_wearable_uid: Option<String>,
     monitor_id: u32,
     use_pii_removal: bool,
+    vad_engine: CliVadEngine,
 ) -> Result<()> {
-    let (whisper_sender, whisper_receiver, whisper_shutdown_flag) = if audio_disabled {
+    let (whisper_sender, whisper_receiver) = if audio_disabled {
         // Create a dummy channel if no audio devices are available, e.g. audio disabled
         let (input_sender, _): (UnboundedSender<AudioInput>, UnboundedReceiver<AudioInput>) =
             unbounded_channel();
         let (_, output_receiver): (
             UnboundedSender<TranscriptionResult>,
             UnboundedReceiver<TranscriptionResult>,
         ) = unbounded_channel();
-        (
-            input_sender,
-            output_receiver,
-            Arc::new(AtomicBool::new(false)),
-        )
+        (input_sender, output_receiver)
     } else {
-        create_whisper_channel(audio_transcription_engine.clone()).await?
+        match vad_engine {
+            CliVadEngine::WebRtc => create_whisper_channel(audio_transcription_engine.clone(), VADDetectionEngine::WebRtc).await?,
+            CliVadEngine::Silero => create_whisper_channel(audio_transcription_engine.clone(), VADDetectionEngine::Silero).await?,
+        }
     };
-    let whisper_sender_clone = whisper_sender.clone();
     let db_manager_video = Arc::clone(&db);
     let db_manager_audio = Arc::clone(&db);
 
-    let is_running_video = Arc::clone(&vision_control);
+    let is_running_video = Arc::clone(&vi   on_control);
```

Comment:
does this compile?

Reply:
this is typing mistake which can be done while pushing i guess , i resolved this.